subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -55,8 +59,10 @@
  * gradle someJavaExecTask --debug-jvm
  * </pre>
  */
-public class JavaExec extends ConventionTask implements JavaExecSpec {
+public class JavaExec extends ConventionTask implements JavaExecSpec, ConfigurableJavaForkOptions {
```

Comment:
Shouldn't `JavaExecSpec` implement this?

Reply:
Good point. That simplifies the code quite a bit.